File: schemas/digital-asset-management/drizzle/collections.ts
// collections: Curated groups of assets for organization and sharing.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, integer, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";
import { workspaces } from "./workspaces";
import { assets } from "./assets";

export const collections = pgTable(
  "collections",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    workspaceId: uuid("workspace_id").notNull().references(() => workspaces.id, { onDelete: "cascade" }),
    name: text("name").notNull(),
    description: text("description"),
    coverAssetId: uuid("cover_asset_id").references(() => assets.id, { onDelete: "set null" }),
    isPublic: boolean("is_public").notNull().default(false),
    assetCount: integer("asset_count").notNull().default(0),
    createdBy: uuid("created_by").notNull().references(() => users.id, { onDelete: "restrict" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_collections_workspace_id").on(table.workspaceId),
    index("idx_collections_created_by").on(table.createdBy),
  ]
);

File: schemas/digital-asset-management/spacetimedb/collections.rs
// collections: Curated groupings of assets within a workspace.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = collections, public)]
pub struct Collection {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub workspace_id: String, // FK → workspaces.id (cascade delete)

    pub name: String,
    pub description: Option<String>,
    pub cover_asset_id: Option<String>, // FK → assets.id (set null on delete)

    pub is_public: bool, // default false
    pub asset_count: i32, // default 0

    #[index(btree)]
    pub created_by: String, // FK → users.id (restrict delete)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
